src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -340,8 +340,11 @@ public void Close()
 
         void Cleanup()
         {
-            Log.Debug("COPY operation ended", _connector.Id);
-            _connector.CurrentCopyOperation = null;
+            Log.Debug("COPY operation ended", _connector?.Id ?? -1);
+
+            if (_connector != null)
```

Comment:
I'm not entirely sure, but I think we've seen issues around concurrency while cleanup (I can look for issues) - I think it's a good idea to make this method as thread-safe as possible. This would mean capturing `_connector` in the stack and checking that (since it may change between the check and the assignment of `CurrentCopyOperation`).